.github/workflows/build.yaml

```diff
@@ -0,0 +1,36 @@
+---
+name: build
+
+on:
+  push:
+    branches: ["master"]
```

Comment:
we dont need to build on push action, as this is for testing, Lets do it only on pull requests